### PR TITLE
Step by step Example CIFAR10 part 2 : SAG Example

### PR DESCRIPTION
**Pull request by chesterxgchen**
### Description
add CIFAR10 SAG pytorch example

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/hello-world/step-by-step/cifar10/sag/sag.ipynb [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    "\n",
+    "## FedAvg with SAG\n",
+    "<img src=\"fed_avg.png\" alt=\"FedAvg\" width=50% height=45% /> <img src=\"sag.png\" alt=\"Scatter and Gather\" width=40% height=40% />\n",
+    "\n",
+    "The Fed Avg aggregation is done on the server side, its weighted on the number of training steps on each client\n",
+    " \n",
+    "## Convert training code to federated learning training code\n",
+    "\n",
+    "We will use the original [Training a Classifer](https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html) example\n",
+    "in pytorch as the code base. The cleanup code (remove comments etc.) can be found in [here](../code/dl/train.py)\n",
```
Comment (YuanTingHsieh) [inline examples/hello-world/step-by-step/cifar10/sag/sag.ipynb:26]: we can consolidate this script in later PR when this one is merged.

File: examples/hello-world/step-by-step/cifar10/sag/sag.ipynb [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+   "id": "6667a9be-a911-4485-a3cb-cc2067852545",
+   "metadata": {},
+   "source": [
+    "## Job Folder and Configurations\n",
+    "\n",
+    "Now we need to setup the configurations for server and clients and constructure Job folder NVFLARE needed to run. We can do this using NVFLARE job CLI. You can study the [Job CLI tutorials](https://github.com/NVIDIA/NVFlare/blob/main/examples/tutorials/job_cli.ipynb) later with all the details. But for now, you can just use the following commands\n",
+    "\n",
+    "* First Let's explain the available job templates\n",
+    "\n",
+    "make sure we know where to find the job templates. If have already set NVFLARE_HOME=<NVFLARE git clone directory> then, you can skipt the folllowing step. \n"
```
Comment (YuanTingHsieh) [inline examples/hello-world/step-by-step/cifar10/sag/sag.ipynb:92]: This sentence "* First Let's explain the available job templates\n" is not consistent with the paragraph above and below.

**Review by YuanTingHsieh — COMMENTED** [review]
Mostly good, leave some comments

**Review by YuanTingHsieh — DISMISSED** [review]
LGTM, please fix the format errors.

**Review by IsaacYangSLA — APPROVED** [review]
Still some CI failures.  Other than that, it's good.

**Pull request merged**